### PR TITLE
Feature/rest apis

### PR DESCRIPTION
**Pull request by fisherstevenk**
Handles reward listing and claiming and the logo (+start of the historical page)

### CODE REVIEW

File: web/src/app/common/services/rest.service.ts [hunk 1]
```diff
@@ -30,34 +32,24 @@ export class RestService {
...
-            .post(url, globalThis.wallet, {headers: headers}))
-            .then(res => this.processResponse(res))
-            .catch(this.handleError);
     }
 
-    public signTx(signature: any, data: any) {
+    public signAndFinalizeTx(signature: any, data: any) {
         const headers = new HttpHeaders().set('Content-Type', 'application/json').set('authorization', 'Eevoo0aemah1ohY6Oheehee4ivahR5ae');
-        const url = '/rwd/multisig/finalize/6/sporwc/' + data.id;
+        const url = '/rwdbuild/multisig/finalize/1/sporwc/' + data.id;
```
Comment (Sbcdn) [inline web/src/app/common/services/rest.service.ts:49]: Do it here like with the other endpoints end replace 1 with "customerId" then we do not have any hard coded stuff inside. 
Even if we for now just use a global variable we can than be sure that its changing at all sport until we have a good strategy for environment variables

File: system/etc/apache2/sites-available/default-ssl.conf [hunk 2]
```diff
@@ -16,7 +16,7 @@
                 SSLCertificateFile      /etc/ssl/certs/apache-server.crt
                 SSLCertificateKeyFile /etc/ssl/private/apache-server.key
 
-                SSLCACertificatePath /etc/apache2/ssl.crt/
+                SSLCACertificateFile /etc/apache2/ssl.crt/ca.crt
                 SSLVerifyClient optional
```
Comment (Sbcdn) [inline system/etc/apache2/sites-available/default-ssl.conf:20]: Should I add certbot ?

File: web/src/app/common/data/token.ts [hunk 3]
```diff
@@ -1,23 +1,40 @@
...
+    public selected: boolean = false;
 
     constructor(data: any) {
         if (data != null) {
             if (data.tokenname) {
-                this.tokenname = data.tokenname;
+                try {
+                    // Yes, has to be done twice 
+                    // No this is just in the blockchain metadata here just once
+                    this.tokenname = UtilityService.hexToString(data.tokenname);
```
Comment (Sbcdn) [inline web/src/app/common/data/token.ts:16]: I changed this, we need to decode this just once, the metadata (blockchain) are not the same format we get from the backend, its different data. 
Metadata is also encoded in a different (discord was just an example).
I provide you data definitions for the endpoints.

File: web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts [hunk 4]
```diff
@@ -10,76 +10,101 @@ import {Token} from "../../../../common/data/token";
...
-    public submitURL: string = "/rwd/multisig/6/sporwc";
-    public listURL: string = "/rwd/listURL";
-    public tokenCols: any[] = [];
+    public selectedTokens: Map<string, Token> = new Map<string, Token>();
     public claimReturn: string;
     public listReturn: string;
     public walletLoaded: boolean = false;
-    @ViewChild('fileUpload', {static: false}) public fileUpload: any;
+    public initialized: boolean = false;
+    public scanURLs = ["https://explorer.cardano-testnet.iohkdev.io/tx/", "https://explorer.cardano.org/tx/"];
```
Comment (Sbcdn) [inline web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts:30]: I personally like cardanoscan more it has more detailed overview but its just taste

File: web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts [hunk 5]
```diff
@@ -92,13 +117,14 @@ export class RewardsComponent extends NotificationComponent implements OnInit {
                     globalThis.wallet.script.SpoRewardClaim = rwd;
                     globalThis.wallet.script.SpoRewardClaim.recipient_stake_addr = globalThis.wallet.sending_stake_addr;
                     globalThis.wallet.script.SpoRewardClaim.recipient_payment_addr = globalThis.wallet.sending_wal_addrs[0];
-                    this.claimSubscription.unsubscribe();
-                    this.claimSubscription = null;
                     this.infoNotification("Submitting token claim");
-                    this.restService.fakeClaimTokens(this.submitURL)
+                    this.restService.buildTokenClaimTx("1", "sporwc")
```
Comment (Sbcdn) [inline web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts:121]: as mentioned in the other file, lets use a global "customerID" variable until we sorted out the environment variable issue.
Also "sporwc" should go to a variable I think but it does not have to be an environment variable as it is not changing.

**Review by Sbcdn — CHANGES_REQUESTED** [review]
Thank you good work, but some stuff to change :)

File: system/etc/apache2/sites-available/default-ssl.conf (hunk 2 above)
Comment (fisherstevenk) [reply system/etc/apache2/sites-available/default-ssl.conf:20]: Upload the files to the server.  I'll get them in place.  Then we can setup certbot

File: web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts (hunk 4 above)
Comment (fisherstevenk) [reply web/src/app/rewardz/components/dashboard/rewards/rewards.component.ts:30]: I can change it

File: web/src/app/common/data/token.ts (hunk 3 above)
Comment (fisherstevenk) [reply web/src/app/common/data/token.ts:16]: Got it

File: web/src/app/common/services/rest.service.ts (hunk 1 above)
Comment (fisherstevenk) [reply web/src/app/common/services/rest.service.ts:49]: Added two globals in the navbar

        globalThis.customerId = 1;
        globalThis.multiSigType = "sporwc";

**Pull request merged**